pkg/knuu/instance_helper.go

```diff
@@ -173,6 +190,20 @@ func (i *Instance) destroyPod() error {
 		return fmt.Errorf("failed to delete pod: %v", err)
 	}
 
+	// Delete the service account for the pod
+	if err := k8s.DeleteServiceAccount(k8s.Namespace(), i.k8sName); err != nil {
```

Comment:
Is there a reason to not use the instance's public method of DeleteServiceAccount you defined in `k8s_serviceaccount.go`?

Reply:
I'm calling this function here